src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuB2BShoppingServiceImplTest.java

```diff
@@ -36,6 +38,42 @@ public void testCreateRequisitionsFromCxml() throws Exception {
 		assertNotNull(requisitions);
 	}
 
+	public void testCreateRequisitionsFromCxmlWithDuplicateItems() throws Exception {
+		B2BShoppingCart cart = CuB2BShoppingCartFixture.B2B_CART_WITH_DUPLICATE_ITEM.createB2BShoppingCart();
+        cart.addShoppingCartItem(B2BShoppingCartItemFixture.B2B_CART_ITEM_DUPLICATE.createB2BShoppingCartItem());
+
+		List<CuRequisitionDocument> cuRequisitionDocuments = b2bShoppingService.createRequisitionsFromCxml(cart, UserNameFixture.ccs1.getPerson());
+
+		assertFalse(CollectionUtils.isEmpty(cuRequisitionDocuments));
+	}
+
+    public void testCheckRequisitionAccountsAreUnique() throws Exception {
+        B2BShoppingCart cart = CuB2BShoppingCartFixture.B2B_CART_WITH_DUPLICATE_ITEM.createB2BShoppingCart();
+        cart.addShoppingCartItem(B2BShoppingCartItemFixture.B2B_CART_ITEM_DUPLICATE.createB2BShoppingCartItem());
+
+        List<CuRequisitionDocument> cuRequisitionDocuments = b2bShoppingService.createRequisitionsFromCxml(cart, UserNameFixture.ccs1.getPerson());
+        List requisitionAccounts = ((RequisitionItem)cuRequisitionDocuments.get(0).getItems().get(0)).getSourceAccountingLines();
+        requisitionAccounts.add(requisitionAccounts.get(0));
+
+        assertFalse(b2bShoppingService.checkRequisitionAccountsAreUnique(cuRequisitionDocuments.get(0)));
+    }
+
+    public void testCheckRequisitionAccountsAreUniqueMultipleItems() throws Exception {
+        B2BShoppingCart cart = CuB2BShoppingCartFixture.B2B_CART_WITH_DUPLICATE_ITEM.createB2BShoppingCart();
+        cart.addShoppingCartItem(B2BShoppingCartItemFixture.B2B_CART_ITEM_DUPLICATE.createB2BShoppingCartItem());
+
+        List<CuRequisitionDocument> cuRequisitionDocuments = b2bShoppingService.createRequisitionsFromCxml(cart, UserNameFixture.ccs1.getPerson());
+        List<RequisitionItem> requisitionItems = cuRequisitionDocuments.get(0).getItems();
+        assertTrue("Expected more than one RequisitionItem", requisitionItems.size() > 1);
+
+        for (RequisitionItem requisitionItem : requisitionItems){
+            List<PurApAccountingLine> requisitionAccounts = requisitionItem.getSourceAccountingLines();
+            requisitionAccounts.add(requisitionAccounts.get(0));
+        }
+
+        assertFalse(b2bShoppingService.checkRequisitionAccountsAreUnique(cuRequisitionDocuments.get(0)));
```

Comment:
Is there a way to update this test to be sure both validation errors are caught?